src/resources/js/views/events-bar.js

```diff
@@ -0,0 +1,138 @@
+ * @since TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} obj tribe.events.views.manager
+ *
+ * @return {void}
+ */
+( function( $, obj ) {
+	'use strict';
+	var $document = $( document );
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		input: '[data-js="tribe-events-events-bar-input-control-input"]',
+		inputWrapper: '[data-js="tribe-events-events-bar-input-control"]',
+		inputKeywordWrapper: '.tribe-common-c-search__input-control--keyword',
+		inputKeywordWrapperFocus: '.tribe-common-c-search__input-control--keyword-focus',
+		inputLocationWrapper: '.tribe-common-c-search__input-control--location',
+		inputLocationWrapperFocus: '.tribe-common-c-search__input-control--location-focus',
+	};
+
+	/**
+	 * Toggles input class if it has a value
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object of click event
+	 *
+	 * @return {void}
+	 */
+	obj.setInputFocusClass = function( event ) {
+		var $input = event.data.target;
+		var $wrapper = event.data.wrapper;
+
+		// Set the focus class if it has content.
+		$wrapper.toggleClass( event.data.inputClassFocus, '' !== $input.val().trim() );
+	};
+
+	/**
+	 * Bind events for the events bar input, on focus and according to their value.
+	 *
+	 * @since TBD
+	 *
+	 * @param {jQuery} $container jQuery object of container.
+	 * @param {String} inputWrapper selector of the input wrapper.
+	 * @param {String} inputWrapperFocus the focus class we'd like to apply.
+	 *
+	 * @return {void}
+	 */
+	obj.bindEventsInputFocus = function( $container, inputWrapper, inputWrapperFocus ) {
+
+		var $wrapper = $container.find( inputWrapper );
+
+		// Bail in case we dont find the wrapper.
+		if ( ! $wrapper.length ) {
+			return;
+		}
+
+		var $input = $wrapper.find( obj.selectors.input );
+
+		// Bail in case we dont find the input.
+		if ( ! $input.length ) {
+			return;
+		}
+
+		$wrapper.toggleClass( inputWrapperFocus.className(), '' !== $input.val().trim() );
+
+		$input.on( 'change', { target: $input, wrapper: $wrapper, inputClassFocus: inputWrapperFocus.className() }, obj.setInputFocusClass );
+	}
+
+	/**
+	 * Binds events for the events bar change listeners
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object for 'afterSetup.tribeEvents' event
+	 * @param {integer} index jQuery.each index param from 'afterSetup.tribeEvents' event
+	 * @param {jQuery} $container jQuery object of view container
+	 * @param {object} data data object passed from 'afterSetup.tribeEvents' event
+	 *
+	 * @return {void}
+	 */
+	obj.bindEvents = function( event, index, $container, data ) {
+		// Bind event for the keyword input.
+		obj.bindEventsInputFocus( $container, obj.selectors.inputKeywordWrapper, obj.selectors.inputKeywordWrapperFocus );
+
+		// Bind event for the location input.
+		obj.bindEventsInputFocus( $container, obj.selectors.inputLocationWrapper, obj.selectors.inputLocationWrapperFocus );
```

Comment:
We want to make it generic and controlled by a `[data-js]` attribute, which would allow creating more fields without any JS added.

Reply:
We actually need to pass what classes we need to set. If it was on the input we could use something generic, but as we're targeting via wrapper, we'll need to know what wrapper are we on.. location, keyword or whatever, to add the `--focus` modifier class.